crates/oxc_query/src/edges.rs

```diff
@@ -0,0 +1,2168 @@
+            Box::new(std::iter::once(Vertex::Span(
+                v.as_class()
+                    .unwrap_or_else(|| {
+                        panic!("expected to have a class vertex, instead have: {v:#?}")
+                    })
+                    .class
+                    .span,
+            )))
+        })
+    }
+
+    pub(super) fn getter<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        class_fn_edge_implem!(contexts, Get)
+    }
+
+    pub(super) fn method<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        class_fn_edge_implem!(contexts, Method)
+    }
+
+    pub(super) fn name_span<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            #[allow(clippy::option_if_let_else)]
+            if let Some(id) = &v
+                .as_class()
+                .unwrap_or_else(|| panic!("expected to have a class vertex, instead have: {v:#?}"))
+                .class
+                .id
+            {
+                Box::new(std::iter::once(Vertex::Span(id.span)))
+            } else {
+                Box::new(std::iter::empty())
+            }
+        })
+    }
+
+    pub(super) fn property<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            Box::new(
+                v.as_class()
+                    .unwrap_or_else(|| {
+                        panic!("expected to have a class vertex, instead have: {v:#?}")
+                    })
+                    .class
+                    .body
+                    .body
+                    .iter()
+                    .filter_map(|class_el| {
+                        if let ClassElement::PropertyDefinition(property) = class_el {
+                            Some(Vertex::ClassProperty(Rc::new(ClassPropertyVertex {
+                                property,
+                                is_abstract: false,
+                            })))
+                        } else if let ClassElement::TSAbstractPropertyDefinition(def) = class_el {
+                            Some(Vertex::ClassProperty(Rc::new(ClassPropertyVertex {
+                                property: &def.property_definition,
+                                is_abstract: true,
+                            })))
+                        } else {
+                            None
+                        }
```

Comment:
This seems like a `match class_el` with more syntax, consider rewriting.

Reply:
done in [2d29fdf](https://github.com/web-infra-dev/oxc/pull/599/commits/2d29fdf060c20c4ce6f4cfcf0be8a9987603e74d)